### PR TITLE
Disable `datadog.agent.check_status` by default

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?

Disable `datadog.agent.check_status` by default. Adds a setting: `integration_check_status_enabled` to re-enable it.

### Motivation

This service check is not widely used. 

### Describe how to test/QA your changes

1. With the default (`integration_check_status_enabled: false`) run the agent. Visit https://app.datadoghq.com/check/summary?id=datadog.agent.check_status (in the appropriate org) and search for your host. There should be no results. 

2. set `integration_check_status_enabled: true` visit https://app.datadoghq.com/check/summary?id=datadog.agent.check_status and search for your host. You should see results for each check run. Confirm the tag `dd_enable_check_intake:true` is present. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/disable-datadog-agent-check-status-6ed6ca34b74091c0.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+other:
+  - |
+    ``datadog.agent.check_status`` is now disabled by default. It can be 
+    re-enabled with ``integration_check_status_enabled`` set to ``true``
```
Comment (hestonhoffman) [inline releasenotes/notes/disable-datadog-agent-check-status-6ed6ca34b74091c0.yaml:12]:
```suggestion
    ``datadog.agent.check_status`` is now disabled by default. To re-enable, 
    set ``integration_check_status_enabled`` to ``true``.
```

**Review by hestonhoffman — APPROVED** [review]
Left a small release note edit

File: releasenotes/notes/disable-datadog-agent-check-status-6ed6ca34b74091c0.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+other:
+  - |
+    ``datadog.agent.check_status`` is now disabled by default. To re-enable,
```
Comment (vickenty) [inline releasenotes/notes/disable-datadog-agent-check-status-6ed6ca34b74091c0.yaml:11]:
```suggestion
    Service check ``datadog.agent.check_status`` is now disabled by default. To re-enable, 
```

File: pkg/collector/worker/worker.go [hunk 3]
```diff
@@ -148,7 +149,7 @@ func (w *Worker) Run() {
 		if err != nil {
 			log.Errorf("Error getting default sender: %v. Not sending status check for %s", err, check)
 		}
-		serviceCheckTags := []string{fmt.Sprintf("check:%s", check.String())}
+		serviceCheckTags := []string{fmt.Sprintf("check:%s", check.String()), "dd_enable_check_intake:true"}
```
Comment (vickenty) [inline pkg/collector/worker/worker.go:152]: I'm curious, what is the purpose if this new tag?

**Review by vickenty — APPROVED** [review]

File: pkg/collector/worker/worker.go (hunk 3 above)
Comment (gh123man) [reply pkg/collector/worker/worker.go:152]: There is backend work to stop processing this metric. The tag will be used for customers that choose to opt-in to keeping this metric enabled so we can differentiate the traffic from "explicitly opt-in" usage vs old agents sending this metric by default.

**Review by pgimalac — APPROVED** [review]

**Pull request merged**